src/main/server.ts

```diff
@@ -1,17 +1,27 @@
 #!/usr/bin/env node
-const { Logger } = require('@hmcts/nodejs-logging')
+const appInsights = require('applicationinsights');
+const config = require('@hmcts/properties-volume').addTo(require('config'));
+const { Logger } = require('@hmcts/nodejs-logging');
 import * as fs from 'fs'
 import * as https from 'https'
 import * as path from 'path'
 import { app } from './app'
 
-const logger = Logger.getLogger('server')
+const logger = Logger.getLogger('server');
 
 // TODO: set the right port for your application
-const port: number = parseInt(process.env.PORT, 10) || 3000
+const port: number = parseInt(process.env.PORT, 10) || 3000;
+
+if (config.secrets && config.secrets.adoption.AppInsightsInstrumentationKey) {
```

Comment:
I prefer moving secrets stuff to another file just to keep everything in one place and export here something like setSecrets.

Reply:
done